src/plugins/unified_field_list/public/components/field_stats/field_stats.tsx

```diff
@@ -149,18 +153,17 @@ const FieldStatsComponent: React.FC<FieldStatsProps> = ({
 
   const setDataView: typeof changeDataView = useCallback(
     (nextDataView) => {
-      if (!isCanceledRef.current) {
+      if (prevFetchedRef.current !== fieldId) {
```

Comment:
`isCanceledRef` was used to prevent changing the state after the component was unmounted. So by removing it we would run into this issue.

Reply:
Updated here [`2ab4d99` (#147322)](https://github.com/elastic/kibana/pull/147322/commits/2ab4d991e26d302b6865b78967aef8d060957c48)